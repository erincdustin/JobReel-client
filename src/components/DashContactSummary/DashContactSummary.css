.DashContactSummary > h4 {
  text-align: center;
}

.options > ul > li {
  margin: 10px;
  list-style: none;
}

.ul {
  padding-left: 0;
  width: 95%;
  margin: auto;
  height: 100%;
}

.viewAll {
  text-align: center;
}