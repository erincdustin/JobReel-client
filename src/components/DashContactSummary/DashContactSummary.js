import React from 'react';
import { Link } from 'react-router-dom';
import './DashContactSummary.css';
import jobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';

class DashContactSummary extends React.Component {

  static contextType = jobReelContext

  componentDidMount() {
    if (this.context.contacts.length === 0) {
      jobReelApiService.getSavedContacts()
        .then(res => {
          this.context.setContacts(res.contacts);
        })
    }
  }

  renderContactSummaries = () => {
    let contacts = Array.from(this.context.contacts);
    contacts = contacts.sort((a, b) => {
      return new Date(b.date_added) - new Date(a.date_added);
    });
    contacts = contacts.slice(0, 3).map(contact => {
      return (
        <li key={contact.contact_id}>{contact.contact_name} - {contact.company} - {contact.email} - <a href={contact.linkedin}>LinkedIn</a></li>
      )
    })
    return (
      <ul>
        {contacts}
      </ul>
    )
  }
  
  render() {
    return (
      <>
        <div className="DashContactSummary">
          <h4>{this.props.section}</h4>
        </div>
        <div className='options'>
            {this.renderContactSummaries()}
        </div>
        <Link to="/contacts">View all</Link>
      </>
    )
  }
}

export default DashContactSummary;