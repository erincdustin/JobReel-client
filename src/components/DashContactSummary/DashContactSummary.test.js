import React from 'react';
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import DashContactSummary from './DashContactSummary'

describe('DashContactSummary smoke test', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<BrowserRouter><DashContactSummary /></BrowserRouter>, div);
      ReactDOM.unmountComponentAtNode(div);
  })
})