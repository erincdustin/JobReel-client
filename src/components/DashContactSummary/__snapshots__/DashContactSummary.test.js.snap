// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashContactSummary snapshot test renders the UI as expected 1`] = `<DashContactSummary />`;
